types/types.go

```diff
@@ -50,6 +50,63 @@ func (c *Coins) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// Replicating the cosmos-sdk bank module Metadata type
+type DenomMetadata struct {
+	Description string `json:"description"`
+	// DenomUnits represents the list of DenomUnits for a given coin
+	DenomUnits []*DenomUnit `json:"denom_units"`
```

Comment:
Why an array of pointers? This allows `null` as an array element. It looks like in Cosmos SDK this is only a pointer because of code generation from proto.

Is this really what we want @alpe?

Reply:
No need for pointers here. Objects are not modified after loading. Please make it `DenomUnits []DenomUnit`